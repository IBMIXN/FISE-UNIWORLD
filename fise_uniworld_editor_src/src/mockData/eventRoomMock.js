const mockEventRoom = {
  title: "mockTitle",
  eventDate: "2021-03-17T00:00:00.000Z",
  meetingTables: [],
  background: "mockBackground",
  scene: "mockScene",
};

const mockEventRoomUpdate = {
  title: "newMockTitle",
  eventDate: "2021-02-27T12:00:00.000Z",
  meetingTables: [],
  background: "newMockBackground",
  scene: "newMockScene",
};

const mockMeetingTable = {
  title: "mockTitle",
  type: "RoundMeetingTable",
  logoUrl: "https://via.placeholder.com/150",
  zoomUrl: "https://zoom.us/",
  posX: 0,
  posY: 0,
};

export { mockEventRoom, mockEventRoomUpdate, mockMeetingTable };
