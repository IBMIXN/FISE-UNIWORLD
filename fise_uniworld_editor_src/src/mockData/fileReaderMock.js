export default class FileReaderMock {
  onerror = jest.fn();
  onload = jest.fn();
  addEventListener = (_, fn) => fn({ target: { result: {} } });
  readAsDataURL = jest.fn();
}
