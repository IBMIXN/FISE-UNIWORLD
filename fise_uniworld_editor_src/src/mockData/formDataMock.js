export default class FormDataMock {
  append = jest.fn();
}
