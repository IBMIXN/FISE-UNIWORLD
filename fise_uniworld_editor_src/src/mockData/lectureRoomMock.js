const mockLectureRoom = {
  title: "mockTitle",
  module: "mockModule",
  lecturer: "mockLecturer",
  startTime: "2021-03-17T09:00:00.000Z",
  endTime: "2021-03-17T10:00:00.000Z",
  firstSlideUrl: "mockUrl",
  numSlides: 1,
};

const mockLectureRoomUpdate = {
  title: "newMockTitle",
  module: "newMockModule",
  lecturer: "newMockLecturer",
  startTime: "2021-02-27T14:00:00.000Z",
  endTime: "2021-02-27T15:00:00.000Z",
  firstSlideUrl: "newMockUrl",
  numSlides: 3,
};

export { mockLectureRoom, mockLectureRoomUpdate };
