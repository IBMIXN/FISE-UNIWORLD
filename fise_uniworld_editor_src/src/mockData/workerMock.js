// https://github.com/webpack-contrib/worker-loader/issues/10#issuecomment-325977537
module.exports = Object.create(null);
