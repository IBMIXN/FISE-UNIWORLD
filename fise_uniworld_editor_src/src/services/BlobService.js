import { axiosConfig } from "../config";

class BlobService {
  url = "/blobs";

  async get(container) {
    return (await axiosConfig.get(`${this.url}/${container}`)).data.blobs;
  }

  async upload(data, container) {
    const config = {
      headers: {
        "content-type": "multipart/form-data",
      },
    };
    return (await axiosConfig.post(`${this.url}/${container}`, data, config)).data.url;
  }

  async delete(fileName, container) {
    const data = { fileName };
    return await axiosConfig.delete(`${this.url}/${container}`, { data });
  }

  async deleteSlides(fileName, numSlides) {
    const data = { fileName, numSlides };
    return await axiosConfig.delete(`${this.url}/slides`, { data });
  }
}

export default new BlobService();
