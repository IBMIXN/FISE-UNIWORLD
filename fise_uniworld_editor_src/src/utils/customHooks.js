import React from "react";

const useReducerState = (defaultState) => {
  const reducer = (state, newState) => ({ ...state, ...newState });
  return React.useReducer(reducer, defaultState);
};

export { useReducerState };
