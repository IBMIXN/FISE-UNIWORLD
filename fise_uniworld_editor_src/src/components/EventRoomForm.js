import PropTypes from "prop-types";
import { UpdateEventRoomPropType } from "../propTypes/EventRoom";
import React from "react";
import moment from "moment";
import { makeStyles } from "@material-ui/core/styles";
import {
  Grid,
  Typography,
  Card,
  CardActions,
  CardContent,
  Button,
  TextField,
  CircularProgress,
} from "@material-ui/core";
import Autocomplete from "@material-ui/lab/Autocomplete";
import MomentUtils from "@date-io/moment";
import { MuiPickersUtilsProvider, KeyboardDatePicker } from "@material-ui/pickers";
import AddIcon from "@material-ui/icons/Add";
import TableGridEditor from "../components/TableGridEditor";
import FileUpload from "../components/FileUpload";
import { useReducerState } from "../utils/customHooks";
import {
  backgrounds as backgroundPresets,
  scenes,
  findBackground,
  findScene,
  findBackgroundThumbnail,
  findSceneThumbnail,
  isBackgroundPreset,
} from "../utils/presets";
import { getFileNameFromBlobUrl } from "../utils/blobs";
import BlobService from "../services/BlobService";

const useStyles = makeStyles((theme) => ({
  eventInfoPanel: {
    marginTop: theme.spacing(1),
    padding: theme.spacing(2),
  },
  divider: {
    padding: theme.spacing(1),
  },
  centerTableGrid: {
    display: "flex",
    justifyContent: "center",
  },
  createButton: {
    "&:hover": {
      backgroundColor: theme.palette.success.dark,
    },
    backgroundColor: theme.palette.success.main,
    color: "#fff",
  },
  addIcon: {
    color: "#fff",
    marginRight: theme.spacing(2),
  },
  progressLoader: {
    color: "#fff",
    marginRight: theme.spacing(2),
  },
  padTop: {
    paddingTop: theme.spacing(1),
  },
  padRight: {
    paddingRight: theme.spacing(1),
  },
  flexDisplay: {
    display: "flex",
  },
  circle: {
    position: "absolute",
    margin: "auto",
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
    height: "40%",
    width: "70%",
    background: "rgba(166,166,166,0.7)",
    zIndex: 2,
    borderRadius: "50%",
    border: "1px solid #aaa",
  },
}));

const EventRoomForm = ({ onSubmit, submitText, updateEventRoom }) => {
  const classes = useStyles();
  const [isLoading, setLoading] = React.useState(false);
  const eventRoomInitialState = {
    title: "",
    scene: "Default",
    background: "Default",
    eventDate: moment().toISOString(),
    meetingTables: [],
  };
  const [eventRoom, setEventRoom] = useReducerState(eventRoomInitialState);
  const [backgrounds, setBackgrounds] = React.useState([
    ...backgroundPresets.map((background) => background.value),
  ]);

  React.useEffect(() => {
    if (updateEventRoom) {
      setEventRoom(updateEventRoom);
    }
  }, [updateEventRoom, setEventRoom]);

  React.useEffect(() => {
    async function fetchBackgrounds() {
      try {
        const fetchedBackgrounds = await BlobService.get("backgrounds");
        setBackgrounds([
          ...backgroundPresets.map((background) => background.value),
          ...fetchedBackgrounds,
        ]);
      } catch (err) {
        console.log(err);
      }
    }
    fetchBackgrounds();
  }, []);

  const updateMeetingTables = (newTables) => {
    setEventRoom({ meetingTables: newTables });
  };

  const handleSelectScene = (e, value) => {
    if (value !== "Default") {
      setEventRoom({ scene: value, background: "Default" });
    } else {
      setEventRoom({ scene: value });
    }
  };

  const handleSelectBackground = (e, value) => {
    setEventRoom({ background: value });
  };

  const handleChange = (e) => {
    const { name, value } = e.target;
    setEventRoom({
      [name]: value,
    });
  };

  const handleUploadBackground = async (e) => {
    e.preventDefault();
    const file = e.target.files[0];
    if (file && file.name.match(/\.(jpg|jpeg|png|gif)$/)) {
      const formData = new FormData();
      formData.append("image", e.target.files[0]);
      const background = await BlobService.upload(formData, "backgrounds");
      setBackgrounds([...backgrounds, background]);
      setEventRoom({ background });
    }
  };

  const handleDateChange = (val) => {
    setEventRoom({
      eventDate: moment(val).toISOString(),
    });
  };

  const handleClickSubmit = async () => {
    setLoading(true);
    const data = { ...eventRoom };
    data.background = data.background === "none" ? undefined : data.background;
    await onSubmit(data);
    setEventRoom(eventRoomInitialState);
    setLoading(false);
  };

  return (
    <Grid container spacing={2}>
      {" "}
      <Grid item md={12}>
        <Card className={classes.eventInfoPanel}>
          <React.Fragment>
            <CardContent>
              <Grid container spacing={3}>
                <Grid item md={5} xs={12}>
                  <TextField
                    fullWidth
                    disabled={isLoading}
                    id="title"
                    name="title"
                    label="Title"
                    placeholder="Enter the event title"
                    value={eventRoom.title}
                    onChange={handleChange}
                    InputLabelProps={{
                      shrink: true,
                    }}
                  />
                  <MuiPickersUtilsProvider utils={MomentUtils}>
                    <KeyboardDatePicker
                      fullWidth
                      disabled={isLoading}
                      autoOk
                      disableToolbar
                      variant="inline"
                      format="YYYY-MM-DD"
                      margin="normal"
                      id="eventDate"
                      label="Event Date"
                      value={eventRoom.eventDate}
                      onChange={handleDateChange}
                      KeyboardButtonProps={{
                        "aria-label": "change date",
                      }}
                    />
                  </MuiPickersUtilsProvider>
                  <div className={classes.divider} />
                  <Autocomplete
                    fullWidth
                    id="selectScene"
                    disableClearable
                    options={scenes.map((scene) => scene.value)}
                    getOptionLabel={(val) => findScene(val)}
                    value={eventRoom.scene}
                    onChange={handleSelectScene}
                    renderInput={(params) => (
                      <TextField
                        {...params}
                        label="Scene"
                        variant="outlined"
                        InputLabelProps={{
                          shrink: true,
                        }}
                      />
                    )}
                  />
                  <div className={classes.divider} />

                  {eventRoom.scene === "Default" && (
                    <Grid className={classes.flexDisplay} item xs>
                      <Autocomplete
                        fullWidth
                        id="selectBackground"
                        className={classes.padRight}
                        disableClearable
                        options={backgrounds}
                        groupBy={(option) => {
                          if (isBackgroundPreset(option)) {
                            if (option.endsWith(".mp4")) {
                              return "Preset (360° Videos)";
                            } else {
                              return "Preset (360° Images)";
                            }
                          } else {
                            return "Custom";
                          }
                        }}
                        getOptionLabel={(option) =>
                          isBackgroundPreset(option)
                            ? findBackground(option)
                            : getFileNameFromBlobUrl(option)
                        }
                        value={eventRoom.background}
                        onChange={handleSelectBackground}
                        renderInput={(params) => (
                          <TextField
                            {...params}
                            label="Background"
                            variant="outlined"
                            InputLabelProps={{
                              shrink: true,
                            }}
                          />
                        )}
                      />
                      <FileUpload
                        id="backgroundUpload"
                        name="backgroundUpload"
                        onChange={handleUploadBackground}
                        width="120px"
                        accept="image/*"
                      >
                        Upload
                      </FileUpload>
                    </Grid>
                  )}
                  <div className={classes.divider} />
                  <Grid container spacing={4}>
                    <Grid item xs={6}>
                      <Typography variant="subtitle2">Scene Preview</Typography>
                      {eventRoom.scene === "Default" ? (
                        <div style={{ position: "relative" }}>
                          <div className={classes.circle}></div>
                          <img
                            width="100%"
                            alt="background preview"
                            src={
                              isBackgroundPreset(eventRoom.background)
                                ? findBackgroundThumbnail(eventRoom.background)
                                : eventRoom.background
                            }
                          />
                        </div>
                      ) : (
                        <img
                          width="100%"
                          alt="scene preview"
                          src={findSceneThumbnail(eventRoom.scene)}
                        />
                      )}
                    </Grid>
                    <Grid item xs={6}>
                      <Typography variant="subtitle2"> </Typography>
                      <Typography component="h3" variant="subtitle2">
                        Scene:
                      </Typography>
                      <Typography component="h3" variant="caption">
                        {findScene(eventRoom.scene)}
                      </Typography>
                      {eventRoom.scene === "Default" && (
                        <React.Fragment>
                          <Typography component="h3" variant="subtitle2" className={classes.padTop}>
                            Background:
                          </Typography>
                          <Typography component="h3" variant="caption">
                            {isBackgroundPreset(eventRoom.background)
                              ? findBackground(eventRoom.background)
                              : getFileNameFromBlobUrl(eventRoom.background)}
                          </Typography>
                        </React.Fragment>
                      )}
                    </Grid>
                  </Grid>
                </Grid>
                <Grid item md={7} xs={12}>
                  <Typography variant="subtitle2">Meeting Tables</Typography>
                  <div className={classes.divider} />
                  <div className={classes.centerTableGrid}>
                    <TableGridEditor
                      tables={eventRoom.meetingTables}
                      updateMeetingTables={updateMeetingTables}
                    />
                  </div>
                </Grid>
              </Grid>
            </CardContent>
            <CardActions>
              <Button
                id="submitButton"
                className={classes.createButton}
                color="primary"
                variant="outlined"
                size="large"
                onClick={handleClickSubmit}
                disabled={isLoading}
              >
                {isLoading ? (
                  <CircularProgress className={classes.progressLoader} size={20} />
                ) : (
                  <AddIcon className={classes.addIcon} />
                )}
                {submitText}
              </Button>
            </CardActions>
          </React.Fragment>
        </Card>
      </Grid>
    </Grid>
  );
};

EventRoomForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitText: PropTypes.string.isRequired,
  updateEventRoom: UpdateEventRoomPropType,
};

EventRoomForm.defaultProps = {
  updateEventRoom: undefined,
};

export default EventRoomForm;
