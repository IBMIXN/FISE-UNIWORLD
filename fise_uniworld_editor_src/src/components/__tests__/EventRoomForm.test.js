import React from "react";
import { shallow } from "enzyme";
import EventRoomForm from "../EventRoomForm";
import { mockEventRoomUpdate } from "../../mockData/eventRoomMock";
import BlobService from "../../services/BlobService";
import { useReducerState } from "../../utils/customHooks";

jest.mock("../../services/BlobService");
jest.mock("../../utils/customHooks", () => {
  return {
    useReducerState: jest.fn(),
  };
});

describe("EventRoomForm test", () => {
  const eventRoom = {
    title: "",
    scene: "Default",
    background: "Default",
    eventDate: "",
    meetingTables: [],
  };
  const setEventRoom = jest.fn();

  const flushPromises = () => new Promise(setImmediate);
  beforeEach(() => {
    useReducerState.mockImplementation(() => [eventRoom, setEventRoom]);
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  it("should match snapshot", () => {
    const component = shallow(
      <EventRoomForm
        onSubmit={() => {}}
        submitText="mockText"
        updateEventRoom={mockEventRoomUpdate}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it("should render not background autocomplete component if non-default scene is selected", () => {
    useReducerState.mockImplementationOnce(() => [
      { ...eventRoom, scene: "mockScene" },
      setEventRoom,
    ]);
    const component = shallow(
      <EventRoomForm onSubmit={() => {}} submitText="mockText" updateEventRoom={{}} />
    );
    expect(component).toMatchSnapshot();
  });

  it("should fetch backgrounds from blob service on load", () => {
    shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    expect(BlobService.get).toBeCalledWith("backgrounds");
  });

  it("should set eventRoom if updateEventRoom prop exists", () => {
    shallow(
      <EventRoomForm
        onSubmit={() => {}}
        submitText="mockText"
        updateEventRoom={mockEventRoomUpdate}
      />
    );
    expect(setEventRoom).toBeCalledWith(mockEventRoomUpdate);
  });

  it("should not set eventRoom if updateEventRoom prop does not exists", () => {
    shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    expect(setEventRoom).not.toBeCalledWith(mockEventRoomUpdate);
  });

  it("should select background correctly from autocomplete", () => {
    const component = shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockBackground = "mockBackground";
    component.find("#selectBackground").first().simulate("change", null, mockBackground);
    expect(setEventRoom).toBeCalledWith({ background: mockBackground });
  });

  it("should select scene correctly from autocomplete", () => {
    const component = shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockScene = "Default";
    component.find("#selectScene").first().simulate("change", null, mockScene);
    expect(setEventRoom).toBeCalledWith({ scene: mockScene });
  });

  it("should set background to default if non-default scene is selected from autocomplete", () => {
    const component = shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockScene = "mockScene";
    component.find("#selectScene").first().simulate("change", null, mockScene);
    expect(setEventRoom).toBeCalledWith({ scene: mockScene, background: "Default" });
  });

  it("should correctly handle title text change event", () => {
    const component = shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = { target: { name: "title", value: "mockTitle" } };
    component.find("#title").first().simulate("change", mockEvent);
    expect(setEventRoom).toBeCalledWith({ title: mockEvent.target.value });
  });

  it("should correctly handle date change event", () => {
    const component = shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockDate = new Date().toISOString();
    component.find("#eventDate").first().simulate("change", mockDate);
    expect(setEventRoom).toBeCalledWith({ eventDate: mockDate });
  });

  it("should correctly handle upload background", async () => {
    BlobService.upload.mockImplementationOnce(() => Promise.resolve("mockBackground"));

    const component = shallow(<EventRoomForm onSubmit={() => {}} submitText="mockText" />);
    const mockEvent = { target: { files: [{ name: "mockImage.jpg" }] }, preventDefault: () => {} };
    component.find("#backgroundUpload").first().simulate("change", mockEvent);
    await flushPromises();
    expect(BlobService.upload.mock.calls[0][1]).toBe("backgrounds");
    expect(setEventRoom).toBeCalledWith({ background: "mockBackground" });
  });

  it("should correctly handle submit form", async () => {
    const mockSubmit = jest.fn();
    const component = shallow(
      <EventRoomForm onSubmit={mockSubmit} submitText="mockText" updateEventRoom={{}} />
    );
    component.find("#submitButton").first().simulate("click");
    await flushPromises();
    expect(mockSubmit).toBeCalledWith(eventRoom);
    expect(setEventRoom).toBeCalled();
  });

  it("should correctly handle submit form when background set to none", async () => {
    useReducerState.mockImplementationOnce(() => [
      { ...eventRoom, background: "none" },
      setEventRoom,
    ]);
    const mockSubmit = jest.fn();
    const component = shallow(
      <EventRoomForm onSubmit={mockSubmit} submitText="mockText" updateEventRoom={{}} />
    );
    component.find("#submitButton").first().simulate("click");
    await flushPromises();
    expect(mockSubmit).toBeCalledWith({ ...eventRoom, background: undefined });
    expect(setEventRoom).toBeCalled();
  });
});
