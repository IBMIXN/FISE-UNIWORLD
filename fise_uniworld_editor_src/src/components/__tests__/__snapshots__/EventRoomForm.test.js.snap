// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventRoomForm test should match snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
   
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-eventInfoPanel-1"
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={5}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="title"
              label="Title"
              name="title"
              onChange={[Function]}
              placeholder="Enter the event title"
              value=""
            />
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change date",
                  }
                }
                allowKeyboardControl={true}
                autoOk={true}
                disableToolbar={true}
                disabled={false}
                format="YYYY-MM-DD"
                fullWidth={true}
                id="eventDate"
                invalidDateMessage="Invalid Date Format"
                label="Event Date"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                value=""
                variant="inline"
                views={
                  Array [
                    "year",
                    "date",
                  ]
                }
              />
            </MuiPickersUtilsProvider>
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Autocomplete))
              disableClearable={true}
              fullWidth={true}
              getOptionLabel={[Function]}
              id="selectScene"
              onChange={[Function]}
              options={
                Array [
                  "Default",
                  "ConferenceHall",
                  "LaboratoryHall",
                ]
              }
              renderInput={[Function]}
              value="Default"
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Grid))
              className="makeStyles-flexDisplay-9"
              item={true}
              xs={true}
            >
              <WithStyles(ForwardRef(Autocomplete))
                className="makeStyles-padRight-8"
                disableClearable={true}
                fullWidth={true}
                getOptionLabel={[Function]}
                groupBy={[Function]}
                id="selectBackground"
                onChange={[Function]}
                options={
                  Array [
                    "Default",
                    "UclMainQuad",
                    "UclPrintRoomCafe",
                    "Hall",
                    "TimesSquareNewYork",
                    "RoyalOperaHouse",
                    "SydneyOperaHouseOutside",
                    "BritishLibraryOutside",
                    "BritishMuseumOutside",
                    "EiffelTowerSunset",
                    "https://uniworldstorage.blob.core.windows.net/videos/Plaza.mp4",
                    "https://uniworldstorage.blob.core.windows.net/videos/UclTour.mp4",
                    "https://uniworldstorage.blob.core.windows.net/videos/Waterfront.mp4",
                  ]
                }
                renderInput={[Function]}
                value="Default"
              />
              <FileUpload
                accept="image/*"
                id="backgroundUpload"
                name="backgroundUpload"
                onChange={[Function]}
                width="120px"
              >
                Upload
              </FileUpload>
            </WithStyles(ForwardRef(Grid))>
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={4}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={6}
              >
                <WithStyles(ForwardRef(Typography))
                  variant="subtitle2"
                >
                  Scene Preview
                </WithStyles(ForwardRef(Typography))>
                <div
                  style={
                    Object {
                      "position": "relative",
                    }
                  }
                >
                  <div
                    className="makeStyles-circle-10"
                  />
                  <img
                    alt="background preview"
                    width="100%"
                  />
                </div>
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={6}
              >
                <WithStyles(ForwardRef(Typography))
                  variant="subtitle2"
                >
                  Â 
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="subtitle2"
                >
                  Scene:
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="caption"
                >
                  Default
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  className="makeStyles-padTop-7"
                  component="h3"
                  variant="subtitle2"
                >
                  Background:
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="caption"
                >
                  Default
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={7}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              variant="subtitle2"
            >
              Meeting Tables
            </WithStyles(ForwardRef(Typography))>
            <div
              className="makeStyles-divider-2"
            />
            <div
              className="makeStyles-centerTableGrid-3"
            >
              <TableGridEditor
                tables={Array []}
                updateMeetingTables={[Function]}
              />
            </div>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-createButton-4"
          color="primary"
          disabled={false}
          id="submitButton"
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          <Memo([object Object])
            className="makeStyles-addIcon-5"
          />
          mockText
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`EventRoomForm test should render not background autocomplete component if non-default scene is selected 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={2}
>
   
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={12}
  >
    <WithStyles(ForwardRef(Card))
      className="makeStyles-eventInfoPanel-1"
    >
      <WithStyles(ForwardRef(CardContent))>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={5}
            xs={12}
          >
            <WithStyles(ForwardRef(TextField))
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              disabled={false}
              fullWidth={true}
              id="title"
              label="Title"
              name="title"
              onChange={[Function]}
              placeholder="Enter the event title"
              value=""
            />
            <MuiPickersUtilsProvider
              utils={[Function]}
            >
              <PickerWithState
                KeyboardButtonProps={
                  Object {
                    "aria-label": "change date",
                  }
                }
                allowKeyboardControl={true}
                autoOk={true}
                disableToolbar={true}
                disabled={false}
                format="YYYY-MM-DD"
                fullWidth={true}
                id="eventDate"
                invalidDateMessage="Invalid Date Format"
                label="Event Date"
                margin="normal"
                maxDate={2100-01-01T00:00:00.000Z}
                maxDateMessage="Date should not be after maximal date"
                minDate={1900-01-01T00:00:00.000Z}
                minDateMessage="Date should not be before minimal date"
                onChange={[Function]}
                openTo="date"
                value=""
                variant="inline"
                views={
                  Array [
                    "year",
                    "date",
                  ]
                }
              />
            </MuiPickersUtilsProvider>
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Autocomplete))
              disableClearable={true}
              fullWidth={true}
              getOptionLabel={[Function]}
              id="selectScene"
              onChange={[Function]}
              options={
                Array [
                  "Default",
                  "ConferenceHall",
                  "LaboratoryHall",
                ]
              }
              renderInput={[Function]}
              value="mockScene"
            />
            <div
              className="makeStyles-divider-2"
            />
            <div
              className="makeStyles-divider-2"
            />
            <WithStyles(ForwardRef(Grid))
              container={true}
              spacing={4}
            >
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={6}
              >
                <WithStyles(ForwardRef(Typography))
                  variant="subtitle2"
                >
                  Scene Preview
                </WithStyles(ForwardRef(Typography))>
                <img
                  alt="scene preview"
                  src="https://via.placeholder.com/150"
                  width="100%"
                />
              </WithStyles(ForwardRef(Grid))>
              <WithStyles(ForwardRef(Grid))
                item={true}
                xs={6}
              >
                <WithStyles(ForwardRef(Typography))
                  variant="subtitle2"
                >
                  Â 
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="subtitle2"
                >
                  Scene:
                </WithStyles(ForwardRef(Typography))>
                <WithStyles(ForwardRef(Typography))
                  component="h3"
                  variant="caption"
                >
                  mockScene
                </WithStyles(ForwardRef(Typography))>
              </WithStyles(ForwardRef(Grid))>
            </WithStyles(ForwardRef(Grid))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            md={7}
            xs={12}
          >
            <WithStyles(ForwardRef(Typography))
              variant="subtitle2"
            >
              Meeting Tables
            </WithStyles(ForwardRef(Typography))>
            <div
              className="makeStyles-divider-2"
            />
            <div
              className="makeStyles-centerTableGrid-3"
            >
              <TableGridEditor
                tables={Array []}
                updateMeetingTables={[Function]}
              />
            </div>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          className="makeStyles-createButton-4"
          color="primary"
          disabled={false}
          id="submitButton"
          onClick={[Function]}
          size="large"
          variant="outlined"
        >
          <Memo([object Object])
            className="makeStyles-addIcon-5"
          />
          mockText
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
